cmd/swarm/swarm-smoke/upload_and_sync.go

```diff
@@ -107,16 +138,82 @@ func cliUploadAndSync(c *cli.Context) error {
 
 // fetch is getting the requested `hash` from the `endpoint` and compares it with the `original` file
 func fetch(hash string, endpoint string, original []byte, ruid string) error {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
You don't need this, just use right away `context.Background()` in the `spancontext.StartSpan`. I don't see us doing anything with `cancel` in order to create a context for cancellation.